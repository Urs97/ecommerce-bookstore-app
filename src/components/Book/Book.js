import React, { useState } from 'react';
import './Book.scss';

const Book = ({ book }) => {
    const [isHovered, setIsHovered] = useState(false);

    // L is large, M is medium and S is small image format
    const img_url = `http://covers.openlibrary.org/b/olid/${book.cover_edition_key}-L.jpg`;
    const img_alt = `${book.title} book cover`;

    return (
        <div className="book">
            {!isHovered && (
                <img src={img_url} alt={img_alt} className="book-img"
                onMouseEnter={() => setIsHovered(true)}/>
            )}
            {isHovered && (
                <div className="book-img hovered-book-img" onMouseLeave={() => setIsHovered(false)}>
                    <span>Details</span>
                    <button className="button">View More</button>
                </div>
            )}
            <span className="book-title-container"><h4 className="title">{book.title}</h4></span>
            <p>{book.authors[0].name}</p>
            <span className="book-price-container title">${book.price[0]}.{book.price[1]}</span>
            <button className="button">Add to cart</button>
        </div>
    );
};

export default Book;