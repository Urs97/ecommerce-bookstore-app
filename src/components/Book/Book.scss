@import '../../styles/shared.scss';

.book {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background-color: #FFFFFF;
    padding: 1.25em 2em;
    border-radius: $border-radius;

    .book-img {
        width: 12em;
        height: 17em;
        background-color: $blue;
        border-radius: $border-radius;
    }

    .title-container {
        flex-grow: 1;
        display: flex;
        align-items: center;
        margin: 0.75em 0 0.5em 0;

        h4 {
            font-size: 1em;
            text-transform: capitalize;
            text-align: center;
        }
    }

    p {
        font-size: 0.85em;
    }

    .price-container {
        font-size: 1.15em;
        margin-top: 0.5em;
    }

    & button {
        margin-top: 1em;
        padding: 0.7em 1.5em;
        font-size: 0.9em;
    }
}