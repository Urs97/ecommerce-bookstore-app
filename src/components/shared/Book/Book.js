import React from 'react';
import './Book.scss';

function Book() {
    return (
        <div>
            This is a book
        </div>
    )
}

export default Book;