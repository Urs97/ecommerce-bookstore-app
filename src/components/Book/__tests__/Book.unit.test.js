import React from 'react';
import {shallow} from 'enzyme';
import Book from '../Book';

describe('Book', () => {
    test('renders empty Book', () => {
        const wrapper = shallow(
            <Book />
        );
        expect(wrapper).toMatchSnapshot();
    });
});