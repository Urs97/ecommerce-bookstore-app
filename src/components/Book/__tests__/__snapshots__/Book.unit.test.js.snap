// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book renders Book 1`] = `
<div
  className="book"
  onFocus={[Function]}
  tabIndex="0"
>
  <img
    alt="Frankenstein book cover"
    className="book-img"
    onMouseEnter={[Function]}
    src="http://covers.openlibrary.org/b/olid/OL6147070M-L.jpg"
  />
  <span
    className="book-title-container"
  >
    <h4
      className="title"
    >
      Frankenstein
    </h4>
  </span>
  <p
    className="book-author"
  >
    Marry Shelley
  </p>
  <span
    className="book-price-container title"
  >
    $
    94.99
  </span>
  <button
    className="button"
    onBlur={[Function]}
    onClick={[Function]}
  >
    Add to cart
  </button>
</div>
`;
